packages/docs/docs/docs/model/03-encode.md

```diff
@@ -1 +1,152 @@
+`@mondrian-framework/model`:
+
+```ts showLineNumbers
+import { result } from "@mondrian-framework/model"
+```
+
+A `Result<A, E>` is the return type of a function that may either fail with an
+error of type `E`, or succeed producing a value of type `A`.
+
+### Building a `Result`
+
+To build a value of type `Result` you can use the two functions `result.ok`
+and `result.fail`:
+
+```ts showLineNumbers
+const success: Result<number, string> = result.ok(10)
+const failure: Result<number, string> = result.fail("error!")
+```
+
+Consider for example a function that performs safe division, returning
+an error if the dividend is 0 instead of returning `Infinity`. It could be
+implemented using `Result`:
+
+```ts showLineNumbers
+function safeDivide(dividend: number, divisor: number): Result<number, string> {
+    return divisor === 0
+        ? result.fail("Division by 0")
+        : result.ok(dividend / divisor)
+}
+```
+
+### Working with `Result`s
+
+The great advantage provided by the result type is that it _forces_ you to deal
+with the error case, focusing your attention not only on the happy path but also
+on what could go wrong.
+
+To get a value out of a `Result` one must first check wether it is successful or
+not:
+
+```ts showLineNumbers
+function printResult(res: Result<number, string>) {
+    // highlight-start
+    if (res.isOk) {
+    // highlight-end
+        // If `isOk` is true, you can access the `.value` property with type
+        // of the successful result
+        console.log(`success: ${res.value}`)
+    } else {
+        // If `isOk` is false, you can access the `.error` property with type
+        // of the failing result
+        console.log(`error: ${res.error}`)
+    }
+}
+
+printResult(safeDivide(10, 2)) // -> success: 5.0
+printResult(safeDivide(10, 0)) // -> error: "Division by 0"
+```
+
+The `Result` interface also has a lot of utility methods to make it easier to
+work with those, you can have a look at their documentation and examples.
+
+## Working with sensitive data
+
+Encoding can be useful if you need to share data in JSON format, or if you need
+to turn the JSON into any other kind of data structure. However, sometimes you
+may wish to be a bit more nuanced with _what_ data gets included in the final
+encoded value.
+
+Imagine you're working with some sensitive data that you need to hide _before_
+sharing the encoded JSON:
+
+```ts showLineNumbers
+const User = types.Infer<typeof user>
+const user = types.object({
+    name: types.string(),
+    secret: types.string(),
+})
+
+const value = { name: "Giacomo", secret: "..." }
+const json = user.encode(value) // -> { name: "Giacomo", secret: "..." }
+shareJsonValue(json) // Uh oh, we ended up sharing the secret value!
+```
+
+In this example every single field ends up encoded in the final object, so we
+end up sharing the user's secret.
+
+One wai to fix this problem is to remember to remove the sensitive data from
```

Comment:
Maybe mention that sensitive data are always encoded to null with `hide` strategy